src/com/spikes2212/genericsubsystems/drivetrains/HolonomicDrivetrain.java

```diff
@@ -1,7 +1,36 @@
 package com.spikes2212.genericsubsystems.drivetrains;
 
-public abstract class HolonomicDrivetrain extends TankDrivetrain {
-	public abstract void holonomicDrive(double speedY, double speedX); // y-forward/backward
-	// x-left/right
+import java.util.function.Consumer;
+
+import edu.wpi.first.wpilibj.command.Command;
+
+/**
+ * This class represents a type of Drivetrain with a x and y axises which
+ * can be controlled separately.
+ */
+public class HolonomicDrivetrain extends TankDrivetrain {
+
+	private Consumer<Double> controlX;
+	private Consumer<Double> controlY;

```

Comment:
Isn't either X or Y already covered by the drivetrain being a `TankDrivetrain`? Be sure you're not creating a way to control something that's controller by the superclass.